src/main/java/uk/gov/hmcts/reform/civil/model/docmosis/sealedclaim/SealedClaimResponseForm.java

```diff
@@ -0,0 +1,172 @@
+            addRepaymentPlan(caseData, builder, totalAmount);
+        } else if (caseData.isPayBySetDate()) {
+            addPayBySetDate(caseData, builder, totalAmount);
+        }
+    }
+
+    private static void addPayBySetDate(CaseData caseData, SealedClaimResponseForm.SealedClaimResponseFormBuilder builder, BigDecimal totalClaimAmount) {
+        builder.payBy(caseData.getRespondToClaimAdmitPartLRspec().getWhenWillThisAmountBePaid())
+            .amountToPay(totalClaimAmount + "")
+            .whyNotPayImmediately(caseData.getResponseToClaimAdmitPartWhyNotPayLRspec());
+    }
+
+    private static void addPayByDatePayImmediately(SealedClaimResponseForm.SealedClaimResponseFormBuilder builder, BigDecimal totalClaimAmount) {
+        builder.payBy(LocalDate.now()
+                          .plusDays(RespondentResponsePartAdmissionPaymentTimeLRspec.DAYS_TO_PAY_IMMEDIATELY))
+            .amountToPay(totalClaimAmount + "");
+    }
+
+    private static void addRepaymentPlan(CaseData caseData, SealedClaimResponseForm.SealedClaimResponseFormBuilder builder, BigDecimal totalClaimAmount) {
+        if (caseData.getRespondent1RepaymentPlan() != null) {
+            builder.repaymentPlan(RepaymentPlanTemplateData.builder()
+                                      .paymentFrequencyDisplay(caseData.getRespondent1RepaymentPlan().getPaymentFrequencyDisplay())
+                                      .firstRepaymentDate(caseData.getRespondent1RepaymentPlan().getFirstRepaymentDate())
+                                      .paymentAmount(MonetaryConversions.penniesToPounds(caseData.getRespondent1RepaymentPlan().getPaymentAmount()))
+                                      .build())
+                .payBy(caseData.getRespondent1RepaymentPlan()
+                           .finalPaymentBy(totalClaimAmount))
+                .whyNotPayImmediately(caseData.getResponseToClaimAdmitPartWhyNotPayLRspec());
+        }
+    }
+
+    private static void alreadyPaid(CaseData caseData, SealedClaimResponseForm.SealedClaimResponseFormBuilder builder) {
+        RespondToClaim respondToClaim = caseData.getResponseToClaim();
+        builder.whyReject("ALREADY_PAID")
+            .howMuchWasPaid(MonetaryConversions.penniesToPounds(respondToClaim.getHowMuchWasPaid()) + "")
+            .paymentDate(respondToClaim.getWhenWasThisAmountPaid())
+            .paymentHow(respondToClaim.getHowWasThisAmountPaid() == PaymentMethod.OTHER
+                            ? respondToClaim.getHowWasThisAmountPaidOther()
+                            : respondToClaim.getHowWasThisAmountPaid()
+                .getHumanFriendly());
+    }
+
+    private static void addDetailsOnWhyClaimIsRejected(CaseData caseData, SealedClaimResponseForm.SealedClaimResponseFormBuilder builder) {
+        builder.freeTextWhyReject(caseData.getDetailsOfWhyDoesYouDisputeTheClaim())
+            .timelineComments(Optional.ofNullable(caseData.getCaseDataLiP()).map(CaseDataLiP::getTimeLineComment).orElse(
```

Comment:
_Optional.ofNullable(caseData.getCaseDataLiP()_
Separate var for visibility

Reply:
sorry, i do not agree with having variables for readability. in my opinion it makes more noise and more lines of code. https://techbeacon.com/app-dev-testing/why-unnecessary-variables-are-bad-your-code#:~:text=The%20myth%20of%20increased%20readability&text=But%20this%20has%20two%20main,while%20scrolling%20through%20the%20code.